### PR TITLE
feat: listen to package info (for release 34)

### PR DESCRIPTION
**Pull request by nytamin**
This PR implements the listenToPackages functionality (WIP)

* **Other information**:

**Status**

- [ ] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] The functionality has been tested by the PR author
- [x] The functionality has been tested by NRK

### CODE REVIEW

**Comment by Julusian** [comment]
@jstarpl @nytamin I went to have a look at what can be done with the ids, and ended up having a play with the regeneration flow of the callback

By loading the ExpectedPackageDB from the db, we can very quickly discover the rundown this affects. This alone would help with the id issue, as we could either scope the packages to the rundown, or use some additional fields which would remain internal to core to clean up the id matching.

Reading through the rest I had some other thoughts
I don't think we gain anything from the PieceInstances checks, as if there isnt a corresponding piece also listening, then rerunning the blueprints wont do anything. But it is possible for adlibs to be listening for packages.
Also, I wasn't keen on the way the segment ingest was re-run, as some docs were being loaded from the db to figure out what to do before getting into the ingest lock function

I did these changes (untested, so sat on a branch)
https://github.com/nrkno/tv-automation-server-core/compare/feat/listen-to-packageInfo...feat/listen-to-packageInfo-julians

**Comment by Julusian** [comment]
We said earlier that expectedPackages should only be accessible from within the same segment, also we re-run the whole segment blueprints at once. And at the moment, in core we already store the owner PieceId on the expectedPackage. So we know that the pieceId on the package must be from the same segment as the piece which is listening for changes. By adding a 'blueprintPackageId' field, then we have all the data we need to remove the id hack

Another thought, why do we have a separate array of 'listenToPackages' to 'expectedPackages'? Why isnt listen simply a property on the expectedPackages the blueprints output? We don't really care which piece is listening to changes, we only care that something in the blueprints is.
This is the missing piece to make my implementation avoid excessive data fetching

**Review by Julusian — APPROVED** [review]
@sparkpunkd I think this is looking ok still. A couple of small bits that would be good to tidy up, I ignored some bits which I know are fixed in the r36 version

File: meteor/CHANGELOG.md [hunk 1]
```diff
@@ -114,6 +115,9 @@ All notable changes to this project will be documented in this file. See [standa
 * PM: fix quantel types and add GUI settings ([9c4bb37](https://github.com/nrkno/tv-automation-server-core/commit/9c4bb37be6f27beaff9f9db387f357fc5148f509))
 * Register shortcuts according to rank in adlib panels ([982cbd2](https://github.com/nrkno/tv-automation-server-core/commit/982cbd2e1a93e7d0f085ef3ab5a20625171f4cb9))
 * Tests for findLastScriptedPieceOnLayer ([903a2cb](https://github.com/nrkno/tv-automation-server-core/commit/903a2cbae8c698b69352e0b94b497f7d89168455))
+=======
```
[the remark this answers is not in the document]
Comment (sparkpunkd) [reply meteor/CHANGELOG.md:118]: Good spot! I thought I'd got all of those. Thanks!

File: meteor/server/api/ingest/packageInfo.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import { RundownId, Rundowns } from '../../../lib/collections/Rundowns'
+import { SegmentId } from '../../../lib/collections/Segments'
+import { assertNever, lazyIgnore } from '../../../lib/lib'
+import { logger } from '../../logging'
+import { runIngestOperationWithCache } from './lockFunction'
+import { Meteor } from 'meteor/meteor'
+import { regenerateSegmentsFromIngestData } from './generation'
+
+/** to-be @deprecated hack for backwards-compatibility*/
+export function onUpdatedMediaObject(_id: MediaObjId, _newDocument: MediaObject | null) {
```
[the remark this answers is not in the document]
Comment (sparkpunkd) [reply meteor/server/api/ingest/packageInfo.ts]: I have already raised this as a comment on #539. But, as it does nothing, it must be safe to remove from `meteor/api/integration/media-scanner.ts`. Done.

**Pull request merged**